### PR TITLE
STL-2204 create anchor btn and use for share buttons

### PR DESCRIPTION
**Pull request by punknoir101**
# Short Description
- new anchor button component created

# Changes
- adds a new generic anchor button component to use for  all different cases to get rid of the "wild growth"
- use them for share buttons
- next step is to exchange all the other anchor buttons 

# Feedback
- something missing?
Hints: 
- You have to start storybook to see it in action
- the one with the icon and text (play, youtube): hover state for icon has to be set in the parent component where it is used, because colors could be different

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

**Review by VictorDelCampo — CHANGES_REQUESTED** [review]
Good idea with the generic anchor button! I didn't go through the all css definition but I checked it live and the share buttons do not look like as specified:
- Not the same font. Figma says BundesSansWeb with 700 Bold, but this is the topic with do we use the actual BundesSansWeb Bold oder the regular on with 700 weight?
- Font color should be also white for facebook and whatsapp
- Hovering effect :
-- Only working with "Link kopieren"
-- "E-Mail schreiben" does not have any hovering effect at all
-- Both facebook and whatsapp only change the font color to white when hovering


And also the display of the button on mobile versions do not put every button in one line as specified in figma?
image

**Comment by punknoir101** [comment]
> Good idea with the generic anchor button! I didn't go through the all css definition but I checked it live and the share buttons do not look like as specified:
> 
>     * Not the same font. Figma says BundesSansWeb with 700 Bold, but this is the topic with do we use the actual BundesSansWeb Bold oder the regular on with 700 weight?
> 
>     * Font color should be also white for facebook and whatsapp
> 
>     * Hovering effect :
>       -- Only working with "Link kopieren"
>       -- "E-Mail schreiben" does not have any hovering effect at all
>       -- Both facebook and whatsapp only change the font color to white when hovering
> 
> 
> And also the display of the button on mobile versions do not put every button in one line as specified in figma?

Thx.
- we should use BundesSansWebBold
- I will check the hovering effects and responsive behaviour

**Comment by punknoir101** [comment]
I have talked to UX - the style should be okay this way. Possibly there will come breaking points adjustments later. @VictorDelCampo

**Review by SannyNguyenHung — APPROVED** [review]
Looking good to me! Thank you Jördis :+1:

File: webapp/client/src/components/ButtonAnchor.js [hunk 1]
```diff
@@ -168,7 +179,7 @@ ButtonAnchor.propTypes = {
   url: PropTypes.string,
   name: PropTypes.string,
   download: PropTypes.bool,
-  target: PropTypes.string,
+  target: PropTypes.oneOf(["_self", "_blank"]),
```
Comment (SannyNguyenHung) [inline webapp/client/src/components/ButtonAnchor.js:182]: can we add just a boolean "openInANewWindow"?

Comment (punknoir101) [reply webapp/client/src/components/ButtonAnchor.js:182]: In fact there are more possibilities than these both, but I will check the suggestions

Comment (punknoir101) [reply webapp/client/src/components/ButtonAnchor.js]: adjust to only use one prop

**Review by VictorDelCampo — APPROVED** [review]
Great! :)

**Pull request merged**